src/components/elements/inputs/boolean-list.js

```diff
@@ -9,8 +9,16 @@ const isChecked = (values, option) => {
   }
 }
 
-export default class CheckboxList extends Input {
+export default class BooleanList extends Input {
+  isMultiSelect(){
+    return this.props.property.type === 'string[]'
+  }
+
   onChange(e, value) {
+    if (!this.isMultiSelect()){
+      return this.update(value);
+    }
+
     const values = [...this.state.value || []]
```

Comment:
Or we should at least not have to do th ```|| []``` fallback. I think we still may want to create a new array since the existing one is immutable.